apps/frontend/src/utils/math.ts

```diff
@@ -16,15 +16,24 @@ export const toWei = (
   value: BigNumberish,
   unitName: BigNumberish = DEFAULT_UNIT,
 ): BigNumber => {
-  if (isBigNumberish(value)) {
+  // handle BigNumber or Numbers in scientific notation
+  if (isBigNumberish(value) || String(value).search(/e[-+]/) >= 0) {
```

Comment:
I would just extract `String(value).search(/e[-+]/) >= 0` to a separate function (e.g. `isScientificNotation`) so it's self-explanatory.